Assets/Plugins/CountlySDK/Services/DeviceIdCountlyService.cs

```diff
@@ -32,12 +49,19 @@ internal void InitDeviceId(string deviceId = null)
             //User provided DeviceID
             //Generate Random DeviceID
             string storedDeviceId = PlayerPrefs.GetString("DeviceID");
-            DeviceId = !_countlyUtils.IsNullEmptyOrWhitespace(storedDeviceId)
-                ? storedDeviceId
-                : !_countlyUtils.IsNullEmptyOrWhitespace(DeviceId)
-                    ? DeviceId
-                    : !_countlyUtils.IsNullEmptyOrWhitespace(deviceId)
-                        ? deviceId : _countlyUtils.GetUniqueDeviceId();
+            if (!_countlyUtils.IsNullEmptyOrWhitespace(storedDeviceId)) {
+                DeviceId = storedDeviceId;
+            } else {
+                if (!_countlyUtils.IsNullEmptyOrWhitespace(DeviceId)) {
+                    return;

```

Comment:
why is this "return" here? There was none in the original implementation.

Reply:
do i need to add this statement?
DeviceId = DeviceId

Reply:
No, but do it without the "return" statement

Reply:
I have removed the return statement.